src/AiClient.php

```diff
@@ -0,0 +1,421 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WordPress\AiClient;
+
+use Generator;
+use WordPress\AiClient\Messages\DTO\Message;
+use WordPress\AiClient\Messages\DTO\MessagePart;
+use WordPress\AiClient\Operations\DTO\GenerativeAiOperation;
+use WordPress\AiClient\Providers\Contracts\ProviderAvailabilityInterface;
+use WordPress\AiClient\Providers\Models\Contracts\ModelInterface;
+use WordPress\AiClient\Providers\Models\ImageGeneration\Contracts\ImageGenerationModelInterface;
+use WordPress\AiClient\Providers\Models\SpeechGeneration\Contracts\SpeechGenerationModelInterface;
+use WordPress\AiClient\Providers\Models\TextGeneration\Contracts\TextGenerationModelInterface;
+use WordPress\AiClient\Providers\Models\TextToSpeechConversion\Contracts\TextToSpeechConversionModelInterface;
+use WordPress\AiClient\Providers\ProviderRegistry;
+use WordPress\AiClient\Results\DTO\GenerativeAiResult;
+use WordPress\AiClient\Utils\Models;
+use WordPress\AiClient\Utils\PromptNormalizer;
+
+/**
+ * Main AI Client class providing both fluent and traditional APIs for AI operations.
+ *
+ * This class serves as the primary entry point for AI operations, offering:
+ * - Fluent API for easy-to-read chained method calls
+ * - Traditional API for array-based configuration (WordPress style)
+ * - Integration with provider registry for model discovery
+ *
+ * @since n.e.x.t
+ *
+ * @phpstan-import-type MessageArrayShape from Message
+ *
+ * phpcs:ignore Generic.Files.LineLength.TooLong
+ * @phpstan-type Prompt string|MessagePart|Message|MessageArrayShape|list<string|MessagePart|MessageArrayShape>|list<Message>
+ */
+class AiClient
+{
+    /**
+     * @var ProviderRegistry|null The default provider registry instance.
+     */
+    private static ?ProviderRegistry $defaultRegistry = null;
+
+    /**
+     * Gets the default provider registry instance.
+     *
+     * @since n.e.x.t
+     *
+     * @return ProviderRegistry The default provider registry.
+     */
+    public static function defaultRegistry(): ProviderRegistry
+    {
+        if (self::$defaultRegistry === null) {
+            self::$defaultRegistry = new ProviderRegistry();
+        }
+
+        return self::$defaultRegistry;
+    }
+
+    /**
+     * Sets the default provider registry instance.
+     *
+     * @since n.e.x.t
+     *
+     * @param ProviderRegistry $registry The provider registry to set as default.
+     */
+    public static function setDefaultRegistry(ProviderRegistry $registry): void
```

Comment:
I sort of get why this might make sense, but do we have a concrete need for this? I could see the ability of _replacing_ the default registry creating a bunch of unexpected situations.

I would much rather prefer starting just with a single default registry and allowing to create custom registries (we could even have a `PromptBuilder::usingRegistry(ProviderRegistry $registry)` method). We could consider allowing to replace the default registry later, but I'm not sure it's worth the risk. In any case I'd prefer to discuss this in its own issue.

Reply:
@Ref34t Per Felix's suggestion, let's remove this for now. 😄